plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeTypeMapping.java

```diff
@@ -0,0 +1,345 @@
+    // no DST in 1970, but has DST in later years (e.g. 2018)
+    private final ZoneId vilnius = ZoneId.of("Europe/Vilnius");
+    // minutes offset change since 1970-01-01, no DST
+    private final ZoneId kathmandu = ZoneId.of("Asia/Kathmandu");
+
+    @BeforeClass
+    public void setUp()
+    {
+        checkState(jvmZone.getId().equals("America/Bahia_Banderas"), "This test assumes certain JVM time zone");
+        checkIsGap(jvmZone, LocalDate.of(1970, 1, 1));
+        checkIsGap(vilnius, LocalDate.of(1983, 4, 1));
+        verify(vilnius.getRules().getValidOffsets(LocalDate.of(1983, 10, 1).atStartOfDay().minusMinutes(1)).size() == 2);
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        snowflakeServer = new TestingSnowflakeServer();
+        return createSnowflakeQueryRunner(
+                snowflakeServer,
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableList.of());
+    }
+
+    @Test
+    public void testBoolean()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("boolean", "true", BOOLEAN, "BOOLEAN '1'")
+                .addRoundTrip("boolean", "false", BOOLEAN, "BOOLEAN '0'")
+                .addRoundTrip("boolean", "NULL", BOOLEAN, "CAST(NULL AS BOOLEAN)")
+                .execute(getQueryRunner(), snowflakeCreateAndInsert("tpch.test_boolean"))
+                .execute(getQueryRunner(), trinoCreateAsSelect("tpch.test_boolean"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("tpch.test_boolean"));
+    }
+
+    @Test(dataProvider = "snowflakeIntegerTypeProvider")
+    public void testInteger(String inputType)
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip(inputType, "-9223372036854775808", BIGINT, "-9223372036854775808")
+                .addRoundTrip(inputType, "9223372036854775807", BIGINT, "9223372036854775807")
+                .addRoundTrip(inputType, "0", BIGINT, "CAST(0 AS BIGINT)")
+                .addRoundTrip(inputType, "NULL", BIGINT, "CAST(NULL AS BIGINT)")
+                .execute(getQueryRunner(), snowflakeCreateAndInsert("tpch.integer"));
+    }
+
+    @DataProvider
+    public Object[][] snowflakeIntegerTypeProvider()
+    {
+        // INT , INTEGER , BIGINT , SMALLINT , TINYINT , BYTEINT, DECIMAL , NUMERIC are aliases for NUMBER(38, 0) in snowflake
+        // https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#int-integer-bigint-smallint-tinyint-byteint
+        return new Object[][] {
+                {"INT"},
+                {"INTEGER"},
+                {"BIGINT"},
+                {"SMALLINT"},
+                {"TINYINT"},
+                {"BYTEINT"},
+        };
+    }
+
+    @Test
+    public void testDecimal()
+    {
+        // TODO: Still can't pass the test with a long value
+        // Snowflake will convert DECIMAL to BIGINT, except DECIMAL with scale > 0
```

Comment:
Please disable the behavior in JDBC connection property and then fix some test cases.

Reply:
Disabled and fixed